src/input/index.es

```diff
@@ -3,12 +3,24 @@ import PropTypes from 'prop-types';
 import omit from 'lodash/omit';
 import classNames from 'classnames';
 
+import { getUniqueID } from './utils';
 import TextInputComponent from './base';
 
 
 class Input extends TextInputComponent {
+  componentDidMount() {
+    super.componentDidMount();
+
+    if (this.props.disableAutoComplete) {
```

Comment:
only check props in render, it changes over time

would be also acceptable to put `id` in this._ since it never changes and call this.forceUpdate() once.